src/API.jl

```diff
@@ -60,12 +60,18 @@ function require_not_empty(pkgs, f::Symbol)
     isempty(pkgs) && pkgerror("$f requires at least one package")
 end
 
+const first_pkg_action = Ref{Bool}(true)
+
 # Provide some convenience calls
 for f in (:develop, :add, :rm, :up, :pin, :free, :test, :build, :status)
     @eval begin
         $f(pkg::Union{AbstractString, PackageSpec}; kwargs...) = $f([pkg]; kwargs...)
         $f(pkgs::Vector{<:AbstractString}; kwargs...)          = $f([PackageSpec(pkg) for pkg in pkgs]; kwargs...)
         function $f(pkgs::Vector{PackageSpec}; kwargs...)
+            if first_pkg_action[]
+                first_pkg_action[] = false
+                Operations.recall_suspended_packages()
```

Comment:
I don't get why this is done here and not in e.g. `_auto_precompile` or `precompile`.

Reply:
It needs to be recalled before any pkg actions that could change the manifest and unsuspend a suspension.
Say a user does this:
- `pkg> add Foo` which installs v0.1.0, which fails during auto-precomp and is suspended
- Restarts julia
- `pkg> up` which updates Foo to v0.1.1, which should fix the package

If we recall state for the first time after the `up` during the auto-precomp, it will restore the old list that says `Foo` should be ignored.

Basically it needs to be recalled at the activation of the environment, but `Pkg.activate` isn't used to activate the first environment.. that seems to be done in Base, where we can't run `recall_suspended_packages()`

So I tack it onto the first (and only first) time the key Pkg actions are run, and subsequently any time `Pkg.activate` is done

Reply:
OK, this is fixed now. See below